tests/cpu/test_clusterization_resolution.cpp

```diff
@@ -0,0 +1,125 @@
+#include <gtest/gtest.h>
+
+class SurfaceBinningTests
+    : public ::testing::TestWithParam<
+          std::tuple<std::string, std::string, std::string, unsigned int>> {};
+
+// This defines the local frame test suite
+TEST_P(SurfaceBinningTests, Run) {
+
+    std::string detector_file = std::get<0>(GetParam());
+    std::string digi_config_file = std::get<1>(GetParam());
+    std::string data_dir = std::get<2>(GetParam());
+    unsigned int event = std::get<3>(GetParam());
+
+    // Read the surface transforms
+    auto surface_transforms = traccc::read_geometry(detector_file);
+
+    // Read the digitization configuration file
+    auto digi_cfg = traccc::read_digitization_config(digi_config_file);
+
+    // Memory resource used by the EDM.
+    vecmem::host_memory_resource host_mr;
+
+    // Algorithms
+    traccc::clusterization_algorithm ca(host_mr);
+    traccc::spacepoint_formation sf(host_mr);
+
+    // Read the cells from the relevant event file
+    traccc::cell_container_types::host cells_truth =
+        traccc::read_cells_from_event(event, data_dir, traccc::data_format::csv,
+                                      surface_transforms, digi_cfg, host_mr);
+
+    // Get Reconstructed Spacepoints
+    auto measurements_recon = ca(cells_truth);
+    auto spacepoints_recon = sf(measurements_recon);
+
+    // Read the hits from the relevant event file
+    traccc::spacepoint_container_types::host spacepoints_truth =
+        traccc::read_spacepoints_from_event(event, data_dir,
+                                            traccc::data_format::csv,
+                                            surface_transforms, host_mr);
+
+    // Check the size of spacepoints
+    EXPECT_TRUE(spacepoints_recon.size() > 0);
+    EXPECT_EQ(spacepoints_recon.size(), spacepoints_truth.size());
+    EXPECT_TRUE(spacepoints_recon.total_size() > 0);
+    EXPECT_EQ(spacepoints_recon.total_size(), spacepoints_truth.total_size());
+
+    for (std::size_t i = 0; i < spacepoints_recon.size(); i++) {
+        EXPECT_EQ(spacepoints_recon.at(i).header,
+                  spacepoints_truth.at(i).header);
+
+        // Get the spacepoint vectors
+        traccc::spacepoint_collection_types::host& sp_collection_recon =
+            spacepoints_recon[i].items;
+
+        traccc::spacepoint_collection_types::host& sp_collection_truth =
+            spacepoints_truth[i].items;
+
+        EXPECT_EQ(sp_collection_recon.size(), sp_collection_truth.size());
+
+        // Iterate over each spacepoint
+        for (std::size_t j = 0; j < sp_collection_recon.size(); j++) {
+            const auto& sp_recon = sp_collection_recon[j];
+            const auto& sp_truth = sp_collection_truth[j];
+
+            // Make sure that the difference in spacepoint position is less than
+            // 1%
+            EXPECT_TRUE(
+                traccc::getter::norm(sp_recon.global - sp_truth.global) /
+                    traccc::getter::norm(sp_recon.global) <
+                0.01);
```

Comment:
🤔  Interesting. Is a percentage figure really the most adequate here? In reality I would hope that our reconstruction accuracy is some absolute value on the surface of the detectors. And not 0 in one corner, and some larger value in the opposite corner. To me the most physically accurate test here would be to say that the reconstructed position has to be within X um from the true position.

Which I would then express like:

```c++
EXPECT_LT(traccc::getter::norm(sp_recon.global - sp_truth.global), 0.01);
```

Assuming that we want to go for 10 um for instance. Though admittedly at this point we should start using the units from Acts to express these magic numbers more clearly.

Reply:
You are right but let's go with percentage for the moment. Once we add lorentz shift to clusterization algorithm, we can test with expected spatial resolution.